pkg/aggregator/internal/limiter/config.go

```diff
@@ -0,0 +1,47 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package limiter
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// FromConfig builds new Limiter from the configuration.
+func FromConfig(pipelineCount int) *Limiter {
+	return fromConfig(pipelineCount, getCgroupMemoryLimit)
+}
+
+func fromConfig(pipelineCount int, cgroupLimitGetter func() (uint64, error)) *Limiter {
+	limit := 0
+	memoryRatio := config.Datadog.GetFloat64("dogstatsd_context_limiter.cgroup_memory_ratio")
+	bytesPerContext := config.Datadog.GetInt("dogstatsd_context_limiter.bytes_per_context")
+	if memoryRatio > 0 && bytesPerContext > 0 {
+		cgroupLimit, err := cgroupLimitGetter()
+		if err != nil {
+			log.Errorf("dogstatsd context limiter: memory based limit configured, but: %v", err)
+		} else {
+			limit = int(memoryRatio*float64(cgroupLimit)) / bytesPerContext
+			log.Debugf("dogstatsd context limiter: memory limit=%d, ratio=%f, contexts limit=%d", cgroupLimit, memoryRatio, limit)
+		}
+	}
+
+	if limit == 0 {
+		limit = config.Datadog.GetInt("dogstatsd_context_limiter.limit")
```

Comment:
Isn't this contradictory with

```
	config.BindEnvAndSetDefault("dogstatsd_context_limiter.limit", 0)         // 0 = disabled.
```

Reply:
I don't think its contradictory, specifying `dogstatsd_context_limiter.limit==0` will still effectively disable the limiter. This value is passed into the Limiter constructor as `global` and `global <=0` gives a `nil` limiter.

Reply:
👍 Got indeed confused here, hence my other strange comment 😅 Thank you.